components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/interceptors/SubscriberRegistrationInterceptor.java

```diff
@@ -96,7 +106,10 @@ public void handleMessage(Message message) {
                         if (!APIConstants.SUPER_TENANT_DOMAIN.equalsIgnoreCase(tenantDomain)) {
                             loadTenantRegistry();
                         }
-                        apiConsumer.addSubscriber(username, groupId);
+
+                        String organization = (String)message.get(RestApiConstants.ORGANIZATION);
+                        Subscriber addedSubscriber = apiConsumer.addSubscriber(username, groupId, organization);
+                        insertOrganizationSubscriberMapping(apiConsumer, addedSubscriber, organization);
```

Comment:
insertOrganizationSubscriberMapping logic needs to be inside apiConsumer.addSubscriber method.

Reply:
It was separated from the addSubscriber method with the following logic.

As in [1] if there is an instance where the subscriber already exists and he logs in from a different organization, it will invoke the else clause from  [2] and in that instance we will also need to add an entry to the subscriber-org-mapping.

[1] - https://github.com/wso2/carbon-apimgt/pull/10706/files/db3cdc9e982f07e9c7419dbc795f595582504931?file-filters%5B%5D=.java&file-filters%5B%5D=.sql#diff-abfb476e8c84c673da981a0dce331ef6595d5ebd4359091e182a6f47f83b74deR125

[2] - https://github.com/wso2/carbon-apimgt/pull/10706/files/db3cdc9e982f07e9c7419dbc795f595582504931?file-filters%5B%5D=.java&file-filters%5B%5D=.sql#diff-abfb476e8c84c673da981a0dce331ef6595d5ebd4359091e182a6f47f83b74deL81

Reply:
Ok, but at least for adding the subscriber we should have to add the organization underneath isn't it? We cannot have a subscriber without a mapping. If someone calls the apiConsumer.addSubscriber and forget to call the latter insertOrganizationSubscriberMapping it will be a problem

And also, insertOrganizationSubscriberMapping is too technical to expose to the REST API. We can rename it as registerSubscriberInOrganization().